File: frontend/src/services/puzzleService.js
const API_URL = 'http://localhost:8080/api/puzzles';

export const getAllPuzzles = async () => {
  try {
    const response = await fetch(API_URL);
    if (!response.ok) {
      // If puzzle service is disabled (404), return empty array
      if (response.status === 404) {
        console.log('Puzzle service is disabled - this is normal for CAT question mode');
        return [];
      }
      throw new Error(`HTTP error! status: ${response.status}`);
    }
    const data = await response.json();
    return data;
  } catch (error) {
    console.log("Puzzle service not available - using CAT question mode only:", error.message);
    // Return empty array so the frontend doesn't crash
    return [];
  }
};

File: backend/src/main/java/com/puzzlertracker/service/PuzzleService.java
package com.puzzlertracker.service;

import com.puzzlertracker.model.Puzzle;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.ArrayList;

/**
 * Puzzle service - disabled by default for CAT question mode
 * Enable by setting: app.puzzle.service.enabled=true
 */
@Service
@ConditionalOnProperty(name = "app.puzzle.service.enabled", havingValue = "true", matchIfMissing = false)
public class PuzzleService {

    /**
     * Mock implementation - returns empty list when database is not available
     */
    public List<Puzzle> getAllPuzzles() {
        // Return empty list for now - this prevents frontend errors
        return new ArrayList<>();
    }
}
